From 46c51fd1a7e14004123fc3b074a93680af51e7b0 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Tue, 17 Mar 2020 19:46:41 +0800
Subject: [PATCH] kernel/ksysfs.c: Call ksysfs_init() early when
 CONFIG_ROCKCHIP_THUNDER_BOOT=y

configfs_init() depends on ksysfs_init() to init kernel_kobj.
But configfs_init() and ksysfs_init() are both in the same
core_initcall level.

Fix this by move ksysfs_init() to the pure_initcall level.

Change-Id: Iaa23a25a59b3a133f3cecb396fc69dd7e65d45ad
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 kernel/ksysfs.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/kernel/ksysfs.c b/kernel/ksysfs.c
index 46ba853656f6..d5fb460eaea8 100644
--- a/kernel/ksysfs.c
+++ b/kernel/ksysfs.c
@@ -268,4 +268,8 @@ static int __init ksysfs_init(void)
 	return error;
 }
 
+#ifdef CONFIG_ROCKCHIP_THUNDER_BOOT
+pure_initcall(ksysfs_init);
+#else
 core_initcall(ksysfs_init);
+#endif
-- 
2.35.3

